### PR TITLE
[3.9] fix: fix inplace mode delta generator impls don't cleanup standby slot properly

### PR DESCRIPTION
**Pull request by Bodong-Yang**
## Introduction

This PR fixes issues caused by the following edge condition found during bench test:
1. path that previously is a directory is a symlink in the new OTA. 

The cause of the problem is, the cleanup_base and directory checking logic has some problem, causing the standby slot is not properly cleaned up in certain condition.

### CODE REVIEW

File: src/otaclient/create_standby/delta_gen.py [hunk 1]
```diff
@@ -200,6 +218,19 @@ def report_one_file(self, *file_size: int, force_report: bool = False) -> None:
...
 
+class _CheckDirResult(NamedTuple):
+    dir_should_be_removed: bool
+    """Whether this folder presents in the new image."""
+
+    dir_should_be_process: bool
+    """Whether delta generator should look into this folder."""
+
+    dir_should_be_fully_scan: bool
+    """Whether delta generator should scan all files under this folder,
```
Comment (timothylin) [inline src/otaclient/create_standby/delta_gen.py:229]: should we put the comments above the variable?

File: src/otaclient/create_standby/delta_gen.py [hunk 2]
```diff
@@ -211,6 +242,8 @@ class DeltaGenFullDiskScan(_DeltaGeneratorBase):
         Path("/usr"),
         Path("/opt/nvidia"),
         Path("/home/autoware/autoware.proj"),
+        # NOTE: for resume on previous failed inplace update mode
+        Path(cfg.OTA_TMP_STORE),
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply src/otaclient/create_standby/delta_gen.py:246]: I will revert this part.
Let's make another PR for a proper inplace update mode resuming feature.

File: src/otaclient/create_standby/delta_gen.py [hunk 3]
```diff
@@ -664,32 +688,35 @@ def _cleanup_base(self):
...
-                    path=str(canonical_curdir_path)
+            if canonical_curdir_path == CANONICAL_ROOT_P:
+                _cleanup_all_files_under_folder(
+                    delta_src_curdir_path, chain(dirnames, filenames)
                 )
-            ):
+                continue
+
+            # uncondtionally cleanup folders that exceeds maximum folder deepths.
+            if len(canonical_curdir_path.parents) > MAX_FOLDER_DEEPTH:
```
[the remark this answers is not in the document]
Comment (Bodong-Yang) [reply src/otaclient/create_standby/delta_gen.py:704]: Yes, but the worst thing it can be is just we need to download it again.
As the base slot is not guaranteed to be clean(i.e., maybe some places will hold huge number of temporary files, or very deep folders, who knows), for full disk scan, we have to implement this limitation.
